### DISCUSSION TITLE
Is there's a way to add external track to the video?

### DISCUSSION THREAD

**Discussion by holms** [iina/iina#3854] [Q&A]
You'd really stand out from the players out there on OSX. 

**VLC** - this is a huge pain, only possible through advanced file open window, and when attaching additional file, and choosing it in a audio track tab, I have silence, while the track plays through vlc if opened alone. 
**mplayer-cli** - full screen doesn't produce interpolation so this it's completely unwatchable. 
**mplayerX** - doesn't have such feature

**Comment by low-batt** [comment]
Under the `Audio` menu select `Show Quick Settings Panel`. In the panel click on the `Load External Audio Track...` button. Add your audio track. See if that works for you.
![discussion-3854](https://user-images.githubusercontent.com/86170219/177676056-2cbed5f5-556d-4872-8bfb-7b1bb19fcee9.jpg)

**Reply by samtrino** [reply]
It would be great if IINA automatically uses external audio track placed in the same directory and with the same name as the video track, of course with the correct "audio" extension (e.g. somevideo.mp4 and somevideo.m4a)... is this possible? thanks.

**Comment by low-batt** [comment]
Automatically adding audio tracks is probably best entered as an issue (feature request) rather than in a discussion that is not tracked.

IINA does not directly support this, as in providing UI controls for it. However many [mpv](https://mpv.io/) features not exposed by IINA are still available. One way to use them is to add them to the `Additional mpv options` setting found on the `Advanced` tab of IINA's settings. Whenever you modify that setting always restart IINA to activate the new values.

From the `mpv` manual section for the [audio-file-auto](https://mpv.io/manual/stable/#options-no-audio-file-auto) option:
> **--audio-file-auto=<no|exact|fuzzy|all>, --no-audio-file-auto**
>
>    Load additional audio files matching the video filename. The parameter specifies how external audio files are matched.
> -   **no:**	Don't automatically load external audio files (default).
> -   **exact:**	Load the media filename with audio file extension.
> -   **fuzzy:**	Load all audio files containing the media filename.
> -   **all:**	Load all audio files in the current and --audio-file-paths directories.

I think that will do what you want. There are some other options in that section of the manual that can be used to control how that option works

**Reply by samtrino** [reply]
Thank you! I didn't know we could add those options, it worked perfectly well with my files... IINA rocks!

**Reply by low-batt** [reply]
Glad to hear it worked.

I have added issue #4794, a feature request for direct support of this `mpv` setting.
